dv/views/api.py

```diff
@@ -1006,55 +1006,51 @@ class ProjectList(ListAPIView):
     serializer_class = ProjectSerializer
 
     def get_queryset(self):
-        queryset = ProjectAllocation.objects.all()
+        queryset = Project.objects.all()
 
         period = self.request.GET.get('period', DEFAULT_PERIOD)  # used in FE
         period_id = FUNDING_PERIODS_DICT[period]  # used in queries
-        queryset = queryset.filter(funding_period=period_id)
+        queryset = queryset.filter(projectallocation__funding_period=period_id)
 
         programme = self.request.query_params.get('programme', None)
         if programme is not None:
-            queryset = queryset.filter(project__programme_id=programme)
+            queryset = queryset.filter(programme_id=programme)
 
         beneficiary = self.request.query_params.get('beneficiary', None)
         if beneficiary is not None:
-            queryset = queryset.filter(state_id=beneficiary)
+            queryset = queryset.filter(projectallocation__state_id=beneficiary)
 
         fm = FM_REVERSED_DICT.get(self.request.query_params.get('fm', None))
         if fm:
-            queryset = queryset.filter(financial_mechanism=fm)
+            queryset = queryset.filter(projectallocation__financial_mechanism=fm)
 
         programme_area_name = self.request.query_params.get('area', None)
         if programme_area_name:
-            queryset = queryset.filter(programme_area__name=programme_area_name)
+            queryset = queryset.filter(projectallocation__programme_area__name=programme_area_name)
         else:
             # Don't add sector name if programme area is present
             sector_name = self.request.query_params.get('sector', None)
             if sector_name:
-                queryset = queryset.filter(priority_sector__name=sector_name)
+                queryset = queryset.filter(projectallocation__priority_sector__name=sector_name)
```

Comment:
`Project` has a m2m relation to `priority_sectors`, so you can filter by
`priority_sectors__name=sector_name`